File: backend/src/main/java/com/project/entities/OtpVerification.java
package com.project.entities;

import java.time.LocalDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "otp_verifications")
@Getter @Setter
public class OtpVerification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long otpId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private String otpHash;

    private LocalDateTime expiresAt;

    private boolean verified = false;
}

File: backend_dotnet/SafeMind.Core/Entities/OtpVerification.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SafeMind.Core.Entities
{
    [Table("otp_verifications")]
    public class OtpVerification
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long OtpId { get; set; }

        [Required]
        [ForeignKey(nameof(User))]
        [Column("user_id")]
        public long UserId { get; set; }

        public User User { get; set; } = null!;

        public string? OtpHash { get; set; }

        public DateTime ExpiresAt { get; set; }

        public bool Verified { get; set; } = false;
    }
}
